hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/EnrichmentProcessor.java

```diff
@@ -43,13 +56,22 @@ public void process(StructuredTrace trace) {
         trace, enrichmentArrivalTimer, ENRICHMENT_ARRIVAL_TIME);
     AvroToJsonLogger.log(LOG, "Structured Trace before all the enrichment is: {}", trace);
     for (Enricher enricher : enrichers) {
+      FutureTask<Void> task =
+          new FutureTask<>(
+              () -> {
+                applyEnricher(enricher, trace);
+                return null;
+              });
       try {
-        applyEnricher(enricher, trace);
+        executorService.submit(task);
+        task.get(enricherTaskTimeout.toSeconds(), TimeUnit.SECONDS);
```

Comment:
If the goal is to just add a mechanism for timeout of the enrichers, should we use something like https://github.com/google/guava/blob/master/guava/src/com/google/common/util/concurrent/TimeLimiter.java#L258 instead of an executorService

Reply:
The implementation of this interface is same as whats done in the pr https://github.com/google/guava/blob/master/guava/src/com/google/common/util/concurrent/SimpleTimeLimiter.java#L195, and we would need to pass in an executor service as well, so I guess preferring this over the current implementation doesn't add anything.

Reply:
But it removes all the verbosity around handling futures and all. Not a bad option to just use the SimpleTimeLimitter.

Reply:
Though I personally don't mind a little bit of verbosity ;-) , moved to use SimpleTimeLimiter